pkg/collect/host_certificate.go

```diff
@@ -31,43 +55,58 @@ func (c *CollectHostCertificate) IsExcluded() (bool, error) {
 }
 
 func (c *CollectHostCertificate) Collect(progressChan chan<- interface{}) (map[string][]byte, error) {
-	var result = KeyPairValid
+	var results []HostCertCollection
+	var message string
 
-	_, err := tls.LoadX509KeyPair(c.hostCollector.CertificatePath, c.hostCollector.KeyPath)
-	if err != nil {
-		if strings.Contains(err.Error(), "no such file") {
-			result = KeyPairMissing
-		} else if strings.Contains(err.Error(), "PEM inputs may have been switched") {
-			result = KeyPairSwitched
-		} else if strings.Contains(err.Error(), "found a certificate rather than a key") {
-			result = KeyPairSwitched
-		} else if strings.Contains(err.Error(), "private key does not match public key") {
-			result = KeyPairMismatch
-		} else if strings.Contains(err.Error(), "failed to parse private key") {
-			if encrypted, _ := isEncryptedKey(c.hostCollector.KeyPath); encrypted {
-				result = KeyPairEncrypted
+	if c.hostCollector.CertificatePath != "" {
+		results = append(results, HostCertParser(c.hostCollector.CertificatePath))
+	}
+
+	if c.hostCollector.KeyPath != "" {
+		_, err := tls.LoadX509KeyPair(c.hostCollector.CertificatePath, c.hostCollector.KeyPath)
```

Comment:
I don't think we want to care about the private keys. We just want to extract cert meta data

Reply:
Great, I have removed KeyPath, that can simply host collector quite much